### PR TITLE
TRIFFHIR-533: Detect if imported resource exists in a different IG

### PR DESCRIPTION
**Pull request by arianamihailescu**

### CODE REVIEW

File: apps/client/src/app/import/import.component.html [hunk 1]
```diff
@@ -34,7 +34,7 @@ <h1>Import</h1>
         <tr *ngFor="let f of files; let fi = index" [class.invalid]="(f.contentType === 0 || f.contentType === 1) && !f.resource">
           <td>
             {{f.name}}
-            <i *ngIf="f.message" class="fas fa-exclamation" style="color: red;" [ngbTooltip]="f.message"></i>
+            <i *ngIf="f.message" class="fas fa-exclamation" style="color: #ff0000;" [ngbTooltip]="f.message"></i>
```
Comment (seanmcilvenna) [inline apps/client/src/app/import/import.component.html:37]: In future, consider making styles like this a standard CSS class. i.e.
```
i.warning {
  color: #ff0000;
}
```
Not necessary for this PR (unless you want), though.

File: apps/client/src/app/import/import.component.ts [hunk 2]
```diff
@@ -44,6 +35,8 @@ class ImportFileModel {
   public vsBundle?: Bundle;
   public message: string;
   public status: 'add'|'update'|'unauthorized'|'pending'|'unknown' = 'pending';
+  public multiple = false;
```
Comment (seanmcilvenna) [inline apps/client/src/app/import/import.component.ts:38]: Consider if other developers know what  "multiple" represents. Maybe consider name `multipleIgs`...

File: apps/client/src/app/import/import.component.ts [hunk 3]
```diff
@@ -205,7 +198,9 @@ export class ImportComponent implements OnInit {
         if (this.errorMessage === '') {
           this.files.push(importFileModel);
         }
-
+        if (this.configService.project && this.configService.project.implementationGuideId) {
```
Comment (seanmcilvenna) [inline apps/client/src/app/import/import.component.ts:201]: I like the double-check of `if (project)` :thumbsup:

File: apps/client/src/app/import/import.component.ts [hunk 4]
```diff
@@ -232,6 +227,22 @@ export class ImportComponent implements OnInit {
     }
   }
 
+  public async getList(importFileModel) {
+
+    let url = `/api/fhir/${importFileModel.resource.resourceType}`;
+
+    if (importFileModel.resource.id) {
```
Comment (seanmcilvenna) [inline apps/client/src/app/import/import.component.ts:234]: Consider doing 
`if (!importFileModel.resource || !importFileModel.resource.resourceType || !importFileModel.resource.id) return;` at the top of the method.
I believe it is possible that if a file fails to parse into a resource, `importFileModel.resource` could be null. And either way, `resourceType` and `id` are both needed to call the API endpoint.

File: apps/server/src/app/fhir.controller.ts [hunk 5]
```diff
@@ -55,6 +55,36 @@ export class FhirController extends BaseController {
...
+    const currentOptions: AxiosRequestConfig = {
+      url: buildUrl(fhirServerBase, 'ImplementationGuide', null, null, { resource: res }),
+      method: 'GET'
+    };
+
+    // Get the all the implementation guides for that resource
+    try {
+      const getResponse = await this.httpService.request(currentOptions).toPromise();
+      const bundle:Bundle = getResponse.data;
+      if (bundle.entry.length >= 2 || bundle.entry.length === 1 && contextImplementationGuideId !== bundle.entry[0].resource.id) {
```
Comment (seanmcilvenna) [inline apps/server/src/app/fhir.controller.ts:76]: Looks great!

**Review by seanmcilvenna — CHANGES_REQUESTED** [review]

File: apps/server/src/app/fhir.controller.ts (hunk 5 above)
Comment (arianamihailescu) [reply apps/server/src/app/fhir.controller.ts:76]: Thanks! I think I would like to change the method name from listIg to validateUniqueIg

Comment (seanmcilvenna) [reply apps/server/src/app/fhir.controller.ts:76]: I agree. That makes sense.

File: apps/client/src/app/import/import.component.html (hunk 1 above)
Comment (arianamihailescu) [reply apps/client/src/app/import/import.component.html]: Did that

File: apps/client/src/app/import/import.component.ts (hunk 2 above)
Comment (arianamihailescu) [reply apps/client/src/app/import/import.component.ts]: Did that

**Review by seanmcilvenna — APPROVED** [review]

**Pull request merged**